pkg/allocation/converters/converter.go

```diff
@@ -469,3 +486,100 @@ func convertGSAPrioritiesToAllocationPriorities(in []agonesv1.Priority) []*pb.Pr
 	}
 	return out
 }
+
+// convertAllocationCountersToGSACounterActions converts a map of AllocationRequest_Counters to a
+// map of GameServerAllocationSpec CounterActions
+func convertAllocationCountersToGSACounterActions(in map[string]*pb.CounterAction) map[string]allocationv1.CounterAction {
+	out := map[string]allocationv1.CounterAction{}
+	for k, v := range in {
+		ca := allocationv1.CounterAction{}
+
+		if v.Action != nil {
+			action := v.Action.GetValue()
+			ca.Action = &action
+		}
+		if v.Amount != nil {
+			amount := v.Amount.GetValue()
+			ca.Amount = &amount
+		}
+		if v.Capacity != nil {
+			capacity := v.Capacity.GetValue()
+			ca.Capacity = &capacity
+		}
+
+		// TODO: Action and Amount must be used together. Do we want to check that both Action & Amount are not nil,
```

Comment:
Should this logic be in https://github.com/googleforgames/agones/blob/68698827d35faaea5659926bc6312c4e8257b9bd/pkg/apis/allocation/v1/gameserverallocation.go#L513?

Reply:
Agreed this should be checkedin the Validate() (so we can generally assume that the request is valid)

So in this case, to be defensive, we should check both are not nil, as that is the expected incoming request, and we don't want to accidentally have a panic if a bug happens upstream.

Reply:
Added code in Validate.